library/std/src/sys/unix/process/process_common.rs

```diff
@@ -202,6 +239,48 @@ impl Command {
         self.gid
     }
 
+    pub fn get_size(&mut self) -> io::Result<usize> {
+        // Envp size calculation is approximate.
+        let env = &self.env;
+        let problem = &mut self.problem;
+        let env_size = self.env_size.get_or_insert_with(|| {
+            let env_map = env.capture();
+            env_map
+                .iter()
+                .map(|(k, v)| {
+                    os2c(k.as_ref(), problem).to_bytes().len()
+                        + os2c(v.as_ref(), problem).to_bytes().len()
+                        + 2
+                })
+                .sum::<usize>()
+                + (env_map.len() + 1) * mem::size_of::<*const u8>()
+        });
+
+        Ok(self.arg_size + *env_size)
+    }
+
+    pub fn check_size(&mut self, refresh: bool) -> io::Result<bool> {
+        use crate::sys;
+        use core::convert::TryInto;
+        if refresh || self.arg_max.is_none() {
+            let (limit, errno) = unsafe {
+                let old_errno = sys::os::errno();
+                sys::os::set_errno(0);
```

Comment:
Why should we clear existing `errno`?

Reply:
I don't think the no-error case automatically clears errno. Does it?